### PR TITLE
Adds Documentation for Entire Timer Subsystem With Minor Refactors

### PR DESCRIPTION
**Pull request by bobbah**
## About The Pull Request

I've documented the entire timer subsystem

Also if you see something wrong or weird please point it out to me I was documenting as I read through and as such there are a lot of assumptions made

## Why It's Good For The Game

You will now understand the timer subsystem

## Changelog
:cl: bobbahbrown
refactor: Documented timer subsystem, minor refactoring to improve code quality
/:cl:

### CODE REVIEW

**Review by AnturK — DISMISSED** [review]
Good PR

File: code/controllers/subsystem/timer.dm [hunk 1]
```diff
@@ -1,31 +1,50 @@
...
-
-	var/list/bucket_list = list() //list of buckets, each bucket holds every timer that has to run that byond tick.
-
-	var/list/timer_id_dict = list() //list of all active timers assoicated to their timer id (for easy lookup)
-
-	var/list/clienttime_timers = list() //special snowflake timers that run on fancy pansy "client time"
-
+	/// world.time of the first entry in the bucket list, effectively the 'start time' of the current buckets
+	var/head_offset = 0
+	/// Index of the first non-empty bucket
```
Comment (MrStonedOne) [inline code/controllers/subsystem/timer.dm:31]: buckets wrap around now, 
```suggestion
	/// Index of the wrap around pivot for buckets. buckets before this are later running buckets wrapped around from the end of the bucket list.
```

File: code/controllers/subsystem/timer.dm [hunk 2]
```diff
@@ -310,17 +362,35 @@ SUBSYSTEM_DEF(timer)
...
+	/// Unique hash generated when TIMER_UNIQUE flag is present
 	var/hash
+	/// Flags associated with the timer, see _DEFINES/subsystems.dm
 	var/list/flags
-	var/spent = 0 //time we ran the timer.
-	var/name //for easy debugging.
-	//cicular doublely linked list
+	/// Time at which the timer was invoked or destroyed
+	var/spent = 0
+	/// An informative name generated for the timer, useful for bugging
```
Comment (MrStonedOne) [inline code/controllers/subsystem/timer.dm:389]:
```suggestion
	/// An informative name generated for the timer as its representation in strings, useful for debugging
```

**Comment by bobbahbrown** [comment]
Thanks for the review MSO

**Comment by AnturK** [comment]
Conflicts.

**Comment by bobbahbrown** [comment]
@AnturK  fixed

**Pull request merged**